Nodejs/Product/Nodejs/Workspace/PackageJsonScannerFactory.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.NodejsTools.Npm;
+using Microsoft.VisualStudio.Workspace;
+using Microsoft.VisualStudio.Workspace.Debug;
+using Microsoft.VisualStudio.Workspace.Indexing;
+
+namespace Microsoft.NodejsTools.Workspace
+{
+    [ExportFileScanner(
+        ProviderType, "PackageJsonProject",
+        new string[] { NodejsConstants.PackageJsonFile },
+        new Type[] { typeof(IReadOnlyCollection<FileDataValue>), typeof(IReadOnlyCollection<FileReferenceInfo>) })]
+    public sealed class PackageJsonScannerFactory : IWorkspaceProviderFactory<IFileScanner>
+    {
+        private const string ProviderType = "{D5D48E50-C894-4C5E-8666-761E2D7B301E}";
+
+        public IFileScanner CreateProvider(IWorkspace workspaceContext)
+        {
+            return new PackageJsonScanner(workspaceContext);
+        }
+
+        private sealed class PackageJsonScanner : BaseFileScanner, IFileScanner, IFileScannerUpToDateCheck
```

Comment:
>PackageJsonScanner [](start = 29, length = 18)

Is this the only subtype of `BaseFileScanner`?

Reply:
In this PR, it's also used for `.ts` files and `tsconfig.json`. I separated the work to make the reviews easier